### ISSUE TITLE
FONEM rule C-27

### ISSUE THREAD

**Issue by chrislit** [Yomguithereal/talisman#154]
Hi @Yomguithereal, I finally got around to implementing FONEM in Abydos. I found the paper a bit lacking in specifics also, especially with respect to the rule ordering. And I don't have very high confidence that my implementation is entirely correct.
But I do think I spotted one bug in your rule C-27 (unless I misunderstood the rule, since I don't read French). My understanding is that C-27 should change Z to S when preceded by a vowel or between two consonants, but your regexp makes the change when _followed_  by a vowel or between two consonants. 
As a result, I got different results on the inputs OZOUADE and POUYEZ.

**Comment by Yomguithereal** [comment]
Hum... I need to find the paper back from my computer and check that. Thanks @chrislit.

**Comment by Yomguithereal** [comment]
You're right. It seems I messed up. Can you just tell me what you got with OZOUADE and POUYEZ?

**Comment by Yomguithereal** [comment]
This algorithm is very messy though :). I you ever need a better algorithm for French I started phonetic algorithms myself [there](https://github.com/Yomguithereal/phonogram). The French one is very good but quite complex and I need a straightforward rules' format so that it becomes easier to port to other languages. You can test it [here](https://yomguithereal.github.io/phonogram/). But note that only the French and Spanish ones are correct. The other languages are not finished.
